python/ray/worker.py

```diff
@@ -2100,121 +2101,25 @@ def _mode(worker=global_worker):
-            ):
-                raise ValueError(
-                    "The keyword 'max_retries' only accepts 0, -1 or a"
-                    " positive integer"
-                )
-            if max_calls is not None and (
-                not isinstance(max_calls, int) or max_calls < 0
-            ):
-                raise ValueError(
-                    "The keyword 'max_calls' only accepts 0 or a positive integer"
-                )
-            return ray.remote_function.RemoteFunction(
-                Language.PYTHON,
-                function_or_class,
-                None,
-                num_cpus,
-                num_gpus,
-                memory,
-                object_store_memory,
-                resources,
-                accelerator_type,
-                num_returns,
-                max_calls,
-                max_retries,
-                retry_exceptions,
-                runtime_env,
-                placement_group,
-                scheduling_strategy,
-            )
-
-        if inspect.isclass(function_or_class):
-            if num_returns is not None:
-                raise TypeError("The keyword 'num_returns' is not allowed for actors.")
-            if max_retries is not None:
-                raise TypeError("The keyword 'max_retries' is not allowed for actors.")
-            if retry_exceptions is not None:
-                raise TypeError(
-                    "The keyword 'retry_exceptions' is not allowed for actors."
-                )
-            if max_calls is not None:
-                raise TypeError("The keyword 'max_calls' is not allowed for actors.")
-            if max_restarts is not None and (
-                not isinstance(max_restarts, int) or max_restarts < -1
-            ):
-                raise ValueError(
-                    "The keyword 'max_restarts' only accepts -1, 0 or a"
-                    " positive integer"
-                )
-            if max_task_retries is not None and (
-                not isinstance(max_task_retries, int) or max_task_retries < -1
-            ):
-                raise ValueError(
-                    "The keyword 'max_task_retries' only accepts -1, 0 or a"
-                    " positive integer"
-                )
-            return ray.actor.make_actor(
-                function_or_class,
-                num_cpus,
-                num_gpus,
-                memory,
-                object_store_memory,
-                resources,
-                accelerator_type,
-                max_restarts,
-                max_task_retries,
-                runtime_env,
-                concurrency_groups,
-                scheduling_strategy,
-            )
-
-        raise TypeError(
-            "The @ray.remote decorator must be applied to "
-            "either a function or to a class."
+def _make_remote(function_or_class, option_dict):
+    # cleanup "None" value options in the dict
+    option_dict = options.user_input_filter(option_dict)
```

Comment:
Why do we need to do this?

Reply:
This is something messy in the original code: there are two (or three) sets of default values: in some places `None` is the default value to be overrided, in some places `None` is the default value you should not override (`None` has some meanings), in other places the default value is number or string. I do not want to change too much code, so `user_input_filter` is just a workaround to unify them.